### PR TITLE
Adding files for Staubli RX160L variant

### PR DESCRIPTION
**Pull request by muhrix**
I have added the necessary files for the Staubli RX160L variant (longer link 4).
Only after I did this, I thought of using xacro:if to distinguish between them.

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
Hi, thanks for the PR. I'll add some comments in-line.

File: staubli_rx160_support/urdf/rx160l.urdf [hunk 1]
```diff
@@ -0,0 +1,181 @@
+<?xml version="1.0" ?>
```
Comment (gavanderhoorn) [inline staubli_rx160_support/urdf/rx160l.urdf:1]: Please regenerate this urdf using `rosrun xacro xacro -o rx160l.urdf rx160l.xacro`. That should get rid of the comments (`RAL 1028` etc), as well as the commented lines.

Comment (gavanderhoorn) [inline staubli_rx160_support/urdf/rx160l.urdf:1]: Also: all links/joints have a `staubli_` prefix added: is that intentional? It is not present in either the `_macro.xacro` or the top-level `.xacro`.

File: staubli_rx160_support/urdf/rx160l_macro.xacro [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+				<geometry>
+					<mesh filename="package://staubli_rx160_support/meshes/rx160/collision/link_3.stl"/>
+				</geometry>
+			</collision>
+		</link>
+		<link name="${prefix}link_4">
+			<visual>
+				<origin xyz="0 0 0" rpy="0 0 0"/>
+				<geometry>
+					<!-- <mesh filename="package://staubli_rx160_support/meshes/rx160/visual/link_4.stl"/> -->
```
Comment (gavanderhoorn) [inline staubli_rx160_support/urdf/rx160l_macro.xacro]: Please remove commented lines.

Same for the `collision` geometry.

**Comment by gavanderhoorn** [comment]
Again, thanks for the PR. Some other comments:
- not using `xacro:if` is actually preferred: we try to keep the nr of parameters to a minimum as they tend to lead to very messy xacros. The duplication is unfortunate, but for now acceptable
- please move `link_4_l.stl` to a sub dir of the `meshes` dir, like so: `meshes/rx160l/{collision,visual}/link_4.stl`. That would remove the need for the `_l` suffix, and we typically place variant specific meshes in their own subdir of the `meshes` directory. Don't forget to update the `_macro.xacro` and to regenerate the urdf
- you probably have checked this, but are the joint limits really identical to the base RX160? I haven't checked the documentation, but in most cases longer manipulators tend to be slower / have tigher limits.
- could you update `package.xml` with info on the specs (like [here](https://github.com/ros-industrial/staubli/blob/135e2f3559bfdb80ba418b83cdd7bab4167db35d/staubli_rx160_support/package.xml#L13-L16))

Finally, if you could squash all commits into just a single "_rx160: adding support for RX160L_" (or something similar) that would be appreciated.

**Comment by muhrix** [comment]
Hi,

Sorry, I forgot to regenerate the URDF file (I was using a prefix in my tests).
Regarding the joint limits, I was actually just checking the manual to ensure it is correct.

I'll make the changes you suggested and update the PR.
Thanks!

**Comment by muhrix** [comment]
@gavanderhoorn I made some changes, but I didn't manage to get rid of the comments in the URDF files (I tried to regenerate the RX160 and I got the same comments there too).

I checked the manual and the specs are exactly the same for joint limits and angular speed. The RX160L has a longer link 4 (30cm) and hence lower payload, but higher linear speed.

**Comment by gavanderhoorn** [comment]
@muhrix wrote:

> I made some changes, but I didn't manage to get rid of the comments in the URDF files

no, that is actually a feature of newer xacro versions. They preserve comments. I'd forgotten about that.

> (I tried to regenerate the RX160 and I got the same comments there too).

yes, could you revert those changes? I'd like to keep this PR about the RX160L only.

> I checked the manual and the specs are exactly the same for joint limits and angular speed. The RX160L has a longer link 4 (30cm) and hence lower payload, but higher linear speed.

ok. Just wanted to make sure it was correct.

**Comment by muhrix** [comment]
Done.

**Comment by gavanderhoorn** [comment]
Ok, thanks again for the PR. Much appreciated.

+1.

**Pull request merged**